src/crypto/hash-extra-jh.c

```diff
@@ -38,5 +38,6 @@
 
 void hash_extra_jh(const void *data, size_t length, char *hash) {
   int r = jh_hash(HASH_SIZE * 8, data, 8 * length, (uint8_t*)hash);
+  (void) r; // silence -Wunused-variable
```

Comment:
Before anybody mentions inconsistency in the casting style:

This is C, not C++, so `static_cast<>` can't be used.

Reply:
So these are only unused because assert() is being no-op'd. These hash functions don't allocate memory so have no reason to fail. Maybe we can just get rid of `r` and the asserts too?

Reply:
Yeah since `hashbitlen` is a compile-time constant, there should be no reason for `jh_hash` to not return `SUCCESS`

Reply:
Okay changes are pushed for that

Reply:
Hi, sorry for the wishy-washiness but I'm gonna take back what I said and go with your initial patch. Safer to leave the asserts in place so that they're present in Debug builds, and will catch any accidental misuses of the code in the future.

Reply:
Should I make the `assert` statements conditional on whether `NDEBUG` is defined?

Reply:
OK, sounds good.